privy-io/privy-quickstart
# Rainbowkit Refactor

[![Loom Demo](https://user-images.githubusercontent.com/45455218/181747418-bc9ab119-849f-4a21-b41b-feaeb62d23dc.png)]()
Click here for [demo video](https://www.loom.com/share/a06971027a2e4e6796a2e53a63189c1d)

**Context**
Hacked on the privy-quick starter guide during the ETHCC Hack. 
Used Rainbowkit during my hack and refactored this starter guide with it. 

Why: Having a great Developer Experiencer (DX) is crucial and making it inviting as much as possible. 

### Approach / Main files worked on:
- ```App.js```
    - Added the Wagmi + Rainbowkit client as a wrapper
- ```Index.js```
    - No need to update Address as this is tracked by Wagmi library
    - Moved the rendering components into smaller components. 
- ```Globals.css```
    - Added Header Div 
    - Removed the body font. 

### In order to test
- Add in privy API key
- Ensure you have the three fields in your console. Follow the guide here.[]()
- ```yarn```
- ```yarn dev```

###  User Flow Edge Cases to consider / test
**First-time User**
- Connect to Rainbowkit
- Auth with the module
- Add in your user Data
- Button should say “Pushed to Privy”
- Check Privy DB
- Refresh and see if data is updated. 

**Existing User (Connected Wallet)**
- Page should load with all previous data
- Input new Data
- Refresh and see if data is updated. 

**Existing User (Not connected wallet)**
- Connect to Wallet
- Old Data should appear. 
- Input new Data
- Refresh and see if data is updated. 

**Existing User / Disconnect Wallet.** 
- Disconnect Wallet
- Page should reset
- Connect Wallet Again
- Old Data should appear. 
- Input new Data
- Refresh and see if data is updated.

### Next Steps
- Test the several user flows / edge cases 
- Provide feedback 
- Adjust the code accordingly
- Remove the unnecessary comments.
